### ISSUE TITLE
Use .editorconfig instead of ruleset files

### ISSUE THREAD

**Issue by patriksvensson** [cake-build/cake#2834]
We should go all in on `.editorconfig` files with analyzer settings instead of using the old ruleset files.

**Comment by nils-a** [comment]
@patriksvensson are you aware, that moving rules into `.editorconfig` only works for VS2019? Earlier versions only support the Code Style-options in .editorconfig and ignore the rules.

**Comment by patriksvensson** [comment]
@nils-a Yes, but development of Cake is done in VS2019. If someone submits a PR created with earlier versions, it will be caught in CI (warning as errors enabled in release mode).

**Comment by devlead** [comment]
Fixed by #5008
